### PR TITLE
service/presetmap: create or update existent presetmap when creating new presets

### LINKED ISSUE #172: presets: PresetMap not being update when creating a new preset with same presetmap name
Steps to reproduce the bug:

Listing the PresetMap for 1080p_mp4:
```shell
$ curl http://transcoding.media-factory.dev.nyt.net/presetmaps/1080p_mp4 | pjson
{
  "name": "1080p_mp4",
  "output": {
    "extension": "mp4"
  },
  "providerMapping": {
    "encodingcom": "1080p_mp4"
  }
}
```
Creating the preset on zencoder, with the same PresetMap name:
```shell
$ cat mp4_1080p.json
{
  "providers": ["zencoder"],
  "preset": {
    "name": "1080p_mp4",
    "description": "1080p mp4",
    "container": "mp4",
    "rateControl": "VBR",
    "video": {
        "profile": "Main",
        "profileLevel": "3.1",
        "height": "1080",
        "width": "",
        "codec": "h264",
        "bitrate": "3700000",
        "gopSize": "90",
        "gopMode": "fixed",
        "interlaceMode": "progressive"
    },
    "audio": {
        "codec": "aac",
        "bitrate": "64000"
    }
  },
  "outputOptions": {
    "extension": "mp4",
    "label": "1080p"
  }
}
```
```shell
$ curl -X POST -d @mp4_1080p.json http://transcoding.media-factory.dev.nyt.net/presets | pjson
{
  "PresetMap": "",
  "Results": {
    "zencoder": {
      "Error": "",
      "PresetID": "1080p_mp4"
    }
  }
}
```
Listing the PresetMap:
```shell
$ curl http://transcoding.media-factory.dev.nyt.net/presetmaps/1080p_mp4 | pjson
{
  "name": "1080p_mp4",
  "output": {
    "extension": "mp4"
  },
  "providerMapping": {
    "encodingcom": "1080p_mp4"
  }
}
```

### PR DESCRIPTION
**Pull request by flavioribeiro**
this fixes #172

### CODE REVIEW

**Comment by flavioribeiro** [comment]
@marzagao it worked now ^

File: service/preset.go [hunk 1]
```diff
@@ -86,11 +87,37 @@ func (s *TranscodingService) newPreset(r *http.Request) swagger.GizmoJSONRespons
...
-	presetMap.OutputOpts = input.OutputOptions
 
-	presetMap.ProviderMapping = make(map[string]string)
 	output.Results = make(map[string]newPresetOutput)
-	for _, p := range input.Providers {
+
+	// Sometimes we try to create a new preset in a new provider but we already
+	// have the PresetMap stored. We want to update the PresetMap in such cases.
+	presetMap, err = s.db.GetPresetMap(input.Preset.Name)
+	if err == db.ErrPresetMapNotFound {
```
Comment (marzagao) [inline service/preset.go]: There's code below this point that assumes that the presetMap is not nil. (lines 136, 141, etc) So you should probably also handle the case for when another error other than `db.ErrPresetMapNotFound` happens, yes?

Comment (flavioribeiro) [inline service/preset.go]: yep @marzagao thanks for the cr

**Comment by marzagao** [comment]
LGTM. Thanks!

**Pull request merged**